ts/features/it-wallet/screens/presentation/crossdevice/new/ItwPresentationChecksScreen.tsx

```diff
@@ -0,0 +1,104 @@
+import React from "react";
+import * as pot from "@pagopa/ts-commons/lib/pot";
+import { useNavigation } from "@react-navigation/native";
+import { SafeAreaView } from "react-native-safe-area-context";
+import { IOStyles } from "@pagopa/io-app-design-system";
+import { itwPresentationChecks } from "../../../../store/actions/new/itwPresentationActions";
+import { itwPresentationChecksSelector } from "../../../../store/reducers/new/itwPresentationReducer";
+import { useIODispatch, useIOSelector } from "../../../../../../store/hooks";
+import { useOnFirstRender } from "../../../../../../utils/hooks/useOnFirstRender";
+import ItwLoadingSpinnerOverlay from "../../../../components/ItwLoadingSpinnerOverlay";
+import ItwErrorView from "../../../../components/ItwErrorView";
+import { cancelButtonProps } from "../../../../utils/itwButtonsUtils";
+import { ItWalletError } from "../../../../utils/errors/itwErrors";
+import ItwContinueScreen from "../../../../components/ItwResultComponent";
+import I18n from "../../../../../../i18n";
+import { showCancelAlert } from "../../../../utils/alert";
+import ROUTES from "../../../../../../navigation/routes";
+import {
+  AppParamsList,
+  IOStackNavigationProp
+} from "../../../../../../navigation/params/AppParamsList";
+import { ItwParamsList } from "../../../../navigation/ItwParamsList";
+import { rpMock } from "../../../../utils/mocks";
+
+/**
+ * This screen is used to perform different checks before initiating the presentation flow.
+ * It shows a loading spinner while the checks are being performed and then it shows a success screen with the name of the relaying party if the checks are successful.
+ * It shows an error screen if the checks fail.
+ * The view is rendered based on the state of the checks pot.
+ */
+const ItwPresentationChecksScreen = () => {
+  const dispatch = useIODispatch();
+  const checksPot = useIOSelector(itwPresentationChecksSelector);
+  const navigation =
+    useNavigation<IOStackNavigationProp<ItwParamsList & AppParamsList>>();
+
+  /**
+   * Callback to be used in case of cancel button press alert to navigate to the home screen and show a toast.
+   */
+  const alertOnPress = () => {
+    navigation.navigate(ROUTES.MAIN, { screen: ROUTES.MESSAGES_HOME });
```

Comment:
Message home or wallet home?

Reply:
It's not clear from the FIGMA but I'd say it's safe to return to the message screen since we're coming from the qrcode scanning screen.